cmd/kured/Dockerfile.multi

```diff
@@ -0,0 +1,21 @@
+FROM --platform=$BUILDPLATFORM golang:bullseye AS build
+
+ARG TARGETOS
+ARG TARGETARCH
+ARG TARGETVARIANT
+
+ENV GOOS=$TARGETOS
+ENV GOARCH=$TARGETARCH
+ENV GOVARIANT=$TARGETVARIANT
+
+WORKDIR /src
+COPY go.mod go.sum .
+RUN go mod download
+COPY . .
+RUN go list -f '{{join .Deps "\n"}}' ./cmd/kured | grep -v /vendor/ | xargs go list -f '{{if not .Standard}}{{ $dep := . }}{{range .GoFiles}}{{$dep.Dir}}/{{.}} {{end}}{{end}}'
+RUN CGO_ENABLED=0 go build -o cmd/kured/kured cmd/kured/*.go
```

Comment:
@davideshay Can we use a makefile-target here instead for compiling the binaries? This would avoid the duplication of the godeps-stuff. Also the `main.Version` is missing here.

Reply:
The problem with running it via Makefile is that you don't get the advantages of the single run of buildx to auto-generate everything in the platform list, as well as create the multi-arch manifest.  There certainly could be a way to get the benefits of both, but I'm not aware right now of how.

On the main version piece, my thinking was that on-main-push would execute everytime there was a push to main, and that workflow run would create an image like "main-0fe532a1" with the short commit tag. When you create a new tag, that would run on-tag workflow, which I modified to use the version tag, publishing main:1.9.1

Reply:
I didn't mean to change the whole architecture from this PR with buildx, that's all fine. My suggestion is only to create a make target similar to `make cmd/kured/kured` which has no hardcoded GO* env-args regarding the OS and ARCH. Then the godeps line from the Dockerfile can be reused in the makefile. (So only replace L15+16 from the Dockerfile)

Reply:
I think this PR would be ready to merge, when this can be addressed. @davideshay

Reply:
@davideshay kindly ping on this.
If you don't have enough time right now, I think we can do this in a follow-up PR too.